File: echo/echo.py
#!/usr/bin/env python

import sys
import subprocess

def call_echo(arg):
    executable=sys.argv[0]
    subprocess.call([executable, "echo", arg])

if len(sys.argv) == 1:
    call_echo("{")
    call_echo("{}")
    call_echo("@{}")
elif len(sys.argv) > 2 and sys.argv[1] == "echo":
    print(sys.argv[2])

File: echo/echo.go
package main

import (
	"os"
	"os/exec"
	"fmt"
)

func main() {
	if len(os.Args) == 1 {
		call_echo("{")
		call_echo("{}")
		call_echo("@{}")
	} else if len(os.Args) > 2 && os.Args[1] == "echo" {
		fmt.Println(os.Args[2])
	}

}

func call_echo(arg string) {
	executable := os.Args[0]
	cmd := exec.Command(executable, "echo", arg)
	out, err := cmd.CombinedOutput()
	if err != nil {
		fmt.Println(err)
	}
	fmt.Print(string(out))
}